packages/javascript-sdk/vite.config.ts

```diff
@@ -1,19 +1,46 @@
 /// <reference types='vitest' />
 import { defineConfig } from 'vite';
-
-import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
+import dts from 'vite-plugin-dts';
+// import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
+import { copyFileSync } from 'fs';
 
 export default defineConfig({
   root: __dirname,
   cacheDir: '../../node_modules/.vite/packages/javascript-sdk',
 
-  plugins: [nxViteTsPaths()],
-
   // Uncomment this if you are using workers.
   // worker: {
   //  plugins: [ nxViteTsPaths() ],
   // },
-
+  build: {
+    lib: {
+      entry: 'src/index.ts',
+      name: 'javascript-sdk',
+      formats: ['es', 'cjs'],
+      fileName(format, name) {
+        return `${name}.${format === 'cjs' ? 'js' : 'mjs'}`;
```

Comment:
its kind of semantics. its because we have commonjs as the `type` in package.json. I'm not remembering why but there was something with changing it to type module that caused me an issue. if we have mjs, they will automatically be read for esmodules despite the type.

the `type` just will affect how `js` files are interpreted. cjs/mjs will always be respected.